File: 01easy/go/69.go
package easy

func mySqrt(x int) int {
	var res int
	for i:=0; i<=x/2; i++{
		if i*i == x {
			res =  i
			break
		}
		if i*i > x {
			res = i-1
			break
		}
	}
	return res
}

// 二分更快
func mySqrt2(x int) int {
	if x < 2 {
		return x
	}
	l, r := 1, x/2
	for r >= l {
		mid := (l + r) / 2
		if mid * mid == x {
			return mid
		}
		if mid * mid < x {
			l = mid + 1
		} else {
			r = mid - 1
		}
	}
	return r
}

File: 01easy/py/69.py
'''
实现某一整数的平方根
'''

# 使用二分法
class Solution:
    def mySqrt(self, x: int) -> int:
        result = 0
        low, high = 0, x
        while low <= high:
            mid = (low + high) // 2
            if mid * mid == x:
                return mid
            elif mid * mid > x:
                high = mid - 1

            else:
                low = mid + 1
                result = mid

        return result